### PR TITLE
Implement exercise sieve

### PR DESCRIPTION
**Pull request by AtelesPaniscus**

### CODE REVIEW

**Comment by AtelesPaniscus** [comment]
Here is the implementation.  I have followed all four guides as best I can.  I there are things I have got wrong, please help.

File: exercises/sieve/src/example.erl [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+-export([sieve/1, test_version/0]).
+
+sieve(N) when N < 2 ->
+    [];
+sieve(N) ->
+    sieve(lists:seq(2,N), []).
+
+sieve([], P) ->
+    lists:reverse(P);
+sieve(S, P) ->
```
Comment (NobbZ) [inline exercises/sieve/src/example.erl]: Can you please match `[N|S]` here?

File: exercises/sieve/src/example.erl [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    N = hd(S),
+    sieve(lists:filter(fun(X) -> X rem N /= 0 end, S), [N|P]).
```
Comment (NobbZ) [inline exercises/sieve/src/example.erl]: Can you create a variable for the filtered list? In my opinion `A = f(…), g(A, …)` is much easier to read than `g(f(…), …)`.

File: exercises/sieve/src/example.erl (hunk 1 above)
Comment (AtelesPaniscus) [reply exercises/sieve/src/example.erl]: Happy to oblige.  It did occur to me at one point that I should not have to use hd/1 explicitly.

This GitHub UI is confusing me but no more than any 'intuitive' GUI does.  I am sure with time I will get the hang of it but if it has let me use it incorrectly do let me know.

File: exercises/sieve/src/example.erl (hunk 2 above)
Comment (AtelesPaniscus) [reply exercises/sieve/src/example.erl]: Why do I get the feeling when I contribute to open source I am not contributing my code ?

If I were asked to improve the readability of this snippet of Erlang I would remove the lambda function.  Ugly little blighters in every language it seems.  So ugly that for several years I avoided using them. I just could not see the point.  Then I met JS.  So ugly from start to finish that lambda function seem right at home.

So why did I use one here ?  Well, Erlang's syntax is a lot less ugly that JavaScript's but that is not it.  My experience tells me it is just a case of getting use the this ugly syntax:  I'll eventually learn to transcribe the damned things in my head.

Just like spaces really.  People used to tell me to take the space out of my (to me readable) code only to tell me to put them back in.  Every programmer is Goldilocks.

Is there a limit to how many of these I can do for Haktoberfest this year ?

**Pull request merged**